### PR TITLE
[IMPROVEMENT] - Use keychain & app groups to manage user/server data.

### LINKED ISSUE #939: [IMPROVEMENT] - Use keychain to user access data
Keychain provides a way to save user data (Eg. Token, CurrentServer...) more easily of that AsyncStorage and can share data between App Groups (Eg. Share Extension), in future we need to unify Access on RC Native Client and RC ReactNative Client, to make this we need use App Groups and Keychain.

https://github.com/oblador/react-native-keychain

### PR DESCRIPTION
**Pull request by djorkaeffalexandre**
Keychain provides a way to save user data (Eg. Token, CurrentServer...) more easily of that AsyncStorage and can share data between App Groups (Eg. Share Extension), in future we need to unify Access on RC Native Client and RC ReactNative Client, to make this we need use App Groups and Keychain.

https://github.com/oblador/react-native-keychain

Closes #939 .

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -233,16 +238,17 @@ SPEC CHECKSUMS:
...
   Protobuf: 7a877b7f3e5964e3fce995e2eb323dbc6831bb5a
   QBImagePickerController: d54cf93db6decf26baf6ed3472f336ef35cae022
-  React: 76e6aa2b87d05eb6cccb6926d72685c9a07df152
-  react-native-orientation-locker: 132a63bab4dddd2a5709f6f7935ad9676b0af7c5
-  react-native-splash-screen: 200d11d188e2e78cea3ad319964f6142b6384865
-  react-native-webview: f3e28b48461c78db833f727feec08b13285e7b61
-  RNDeviceInfo: 958a1ed6f94e04557b865b8ef848cfc83db0ebba
-  RNImageCropPicker: e608efe182652dc8690268cb99cb5a201f2b5ea3
-  RNScreens: 720a9e6968beb73e8196239801e887d8401f86ed
+  React: 1d605e098d69bdf08960787f3446f0a9dc2e2ccf
```
[the remark this answers is not in the document]
Comment (djorkaeffalexandre) [reply ios/Podfile.lock]: Fixed

**Comment by djorkaeffalexandre** [comment]
@diegolmello Fixed.

File: app/sagas/init.js [hunk 2]
```diff
@@ -14,22 +14,18 @@ import database from '../lib/realm';
...
 	try {
-		const { token, server } = yield all({
-			token: AsyncStorage.getItem(RocketChat.TOKEN_KEY),
-			server: AsyncStorage.getItem('currentServer')
-		});
-
+		const { serversDB } = database.databases;
+		const currentServer = yield serversDB.objects('servers').filtered('currentServer = true');
+		const { id: server = null } = currentServer.length && currentServer[0];
+		const { password: token } = server ? yield Keychain.getInternetCredentials(server) : { password: null };
```
Comment (IlarionHalushka) [inline app/sagas/init.js]: const { password: token = null } = server && yield Keychain.getInternetCredentials(server);

File: app/sagas/selectServer.js [hunk 3]
```diff
@@ -36,8 +37,21 @@ const handleSelectServer = function* handleSelectServer({ server, version, fetch
...
+				// eslint-disable-next-line no-plusplus
+				for (let i = 0; i < servers.length; i++) {
+					serversDB.create('servers', { id: servers[i].id, currentServer: servers[i].id === server }, true);
+				}
+			} catch (e) {
+				log('err_update_current_server ->', e);
+			}
+		});
+
+		const { username: userStringified } = yield Keychain.getInternetCredentials(server);
```
Comment (IlarionHalushka) [inline app/sagas/selectServer.js]: for safety reasons :) `const { username: userStringified = null } = server && yield Keychain.getInternetCredentials(server);`

File: app/views/RoomsListView/ServerDropdown.js [hunk 4]
```diff
@@ -124,7 +124,7 @@ class ServerDropdown extends Component {
 
 		this.close();
 		if (currentServer !== server) {
-			const token = await AsyncStorage.getItem(`${ RocketChat.TOKEN_KEY }-${ server }`);
+			const { password: token } = await Keychain.getInternetCredentials(server);
```
Comment (IlarionHalushka) [inline app/views/RoomsListView/ServerDropdown.js]: same here, assign null as default `token` value

File: app/sagas/init.js (hunk 2 above)
Comment (djorkaeffalexandre) [reply app/sagas/init.js]: nice man!

Comment (djorkaeffalexandre) [reply app/sagas/init.js]: @IlarionHalushka This doesn't works :( because if server fail will return undefined and try to access password on undefined.

**Pull request closed without merging**